index.js

```diff
@@ -1,301 +1,42 @@
-
-  return geographies.map(geo => {
-    const ft = admin.features.find(c => c.properties.ISO_A2.toLowerCase() === geo.iso)
-    const b = ft ? bbox(ft) : null
-
-    return { ...geo, bbox: b }
-  })
-}
-
-// Generate an overview of the charts
-function generateChartMeta (charts, answers) {
-  return charts.map(c => {
-    if (c.type === 'answer') {
-      c['options'] = answers
-        .filter(a => a.indicator === c.indicatorId)
-        .map(a => ({
-          'id': a.id,
-          'label': a.label
-        }))
-    }
-    const { indicatorId, ...d } = c
-    return d
-  })
-}
-
-function noDataWarning (type, geo) {
-  console.log(`${type} - No data for ${geo}`)
-}
+const load = require('./scripts/load-data')
+const generate = require('./scripts/generate-data')
+const utils = require('./scripts/utils');
 
 (async function main () {
   try {
     // Empty the output folder. Create them if they don't exist.
     await ['./output', './output/results'].forEach(f => fs.emptyDirSync(f))
 
-    const topics = await loadTopics()
-    const charts = await loadCSV('./input/definitions/charts.csv')
-    const answers = await loadCSV('./input/definitions/answers.csv')
+    const topics = await load.topics()
+    const charts = await utils.loadCSV('./input/charts.csv')
+    const answers = await utils.loadCSV('./input/answers.csv')
 
-    const geographies = await loadGeographies()
+    const geographies = await load.geographies()
 
     // TODO Handle multiple years
-    const scores = await loadScoreData(2018)
+    const scores = await load.scoreData(2018)
 
     // No need to load multiple years, only the most recent
-    const indicators = [].concat(await loadSubIndicatorData(2018), await loadInvestmentData(2018))
+    const indicators = [].concat(await load.subIndicatorData(2018), await load.investmentData(2018))
 
     // Contains overall and topic scores
-    const resultData = generateResultData(geographies, scores, topics)
+    const resultData = generate.results(geographies, scores, topics)
 
     // Contains scores, and the auxiliary data to build the charts
-    const detailedResultData = generateDetailedResultData(resultData, indicators, charts)
+    const detailedResultData = generate.detailedResults(resultData, indicators, charts)
 
-    const chartMeta = generateChartMeta(charts, answers)
-    const geographyData = await generateGeographyData(geographies)
+    const chartMeta = generate.chartMeta(charts, answers)
+    const geographyData = await generate.geographies(geographies)
 
-    await fs.writeJson('./output/geographies.json', geographyData)
-    await fs.writeJson('./output/chart-meta.json', chartMeta)
-    await fs.writeJson('./output/results.json', resultData)
-    await Promise.all(detailedResultData.map(geo => fs.writeJson(`./output/results/${geo.iso}.json`, geo)))
+    await Promise.all([
+      detailedResultData.map(geo => fs.writeJson(`./output/results/${geo.iso}.json`, geo)),
```

Comment:
`map` returns an array which needs to be flattened out.
```suggestion
      ...detailedResultData.map(geo => fs.writeJson(`./output/results/${geo.iso}.json`, geo)),
```